lib/maze/servlets/servlet.rb

```diff
@@ -62,16 +62,18 @@ def do_POST(request, response)
       rescue JSON::ParserError => e
         msg = "Unable to parse request as JSON: #{e.message}"
         $logger.error msg
-        Server.invalid_requests << {
+        Server.invalid_requests.add({
           reason: msg,
-          request: request
-        }
+          request: request,
+          body: request.body
+        })
       rescue StandardError => e
         $logger.error "Invalid request: #{e.message}"
-        Server.invalid_requests << {
+        Server.invalid_requests.add({
           reason: e.message,
-          request: request
-        }
+          request: request,
+          body: request.body
+        })
```

Comment:
I've got an example of a flaking test that generates invalid requests - with the request body being empty.  Even trying to add `request` to the hash caused it to raise an exception before it could add the request to the list.

I got around this by changing as follows:

```suggestion
          request: request.inspect
        })
```

With this, the request is successfully added to the list - but writing it out after the scenario fails because it doesn't have the expected form.  Perhaps we just need to settle for a more basic logging approach for any invalid requests - where it just dumps out the raw contents?